dashboard/lib/policies/user.rb

```diff
@@ -10,4 +10,18 @@ def self.user_attributes(user)
 
     attributes.merge('authentication_options_attributes' => authentication_options).compact
   end
+
+  # Determines if a user passes some of the criteria to be a verified teacher.
+  # In order to be a verified teacher candidate, the user must:
+  # - Successfully sync a Google Classroom
+  # - Have a google_oauth2 Authentication Option
+  # - Have a non-google/non-gmail email domain attached to that googele_oauth2 authentication option
+  def self.verified_teacher_candidate?(user)
+    google_ao = user.authentication_options.find_by(credential_type: AuthenticationOption::GOOGLE)
+    is_google_email = google_ao.email.end_with?('@gmail.com', '@googlemail.com')
+    if google_ao && !is_google_email
+      return true
+    end
+    return false
```

Comment:
Nitpick since it's probably impossible in the spot where we use the function, but this will error out finding the email property if `google_ao` is nil right? To make the function stand on its own better, you could move the `is_google_email logic` to the right side of the && to avoid an error:
```suggestion
    if google_ao.present? && !google_ao.email.end_with?('@gmail.com', '@googlemail.com')
      return true
    end
    return false
```

Reply:
Yeah, this was indeed faulty, and I fixed it when I added the unit test for this function this morning. If you refresh, you should see the updated code. Let me know if it looks good or if you'd prefer it as you suggested.

Reply:
Your way looks good to me. Just different syntax to accomplish the same thing.